app/commands/github/task/search.rb

```diff
@@ -0,0 +1,85 @@
+module Github
+  class Task::Search
+    include Mandate
+
+    # Use class method rather than constant for
+    # easier stubbing during testing
+    def self.requests_per_page
+      20
+    end
+
+    def initialize(actions: nil, knowledge: nil, areas: nil, sizes: nil, types: nil, repo_url: nil,
+                   track_id: nil, order: nil, page: 1)
+      @actions = actions
+      @knowledge = knowledge
+      @areas = areas
+      @sizes = sizes
+      @types = types
+      @repo_url = repo_url
+      @track_id = track_id
+      @order = order
+      @page = page
+    end
+
+    def call
+      @tasks = Github::Task
+
+      filter_track!
+      filter_repo!
+      filter_actions!
+      filter_knowledge!
+      filter_areas!
+      filter_sizes!
+      filter_types!
+
+      sort!
+      paginate!
```

Comment:
I think either inline pagination here, or return `@tasks` at the end. The reason is that we might end up having an option not to paginate in search (for example, when we want to see how many issues are open for a filter and want to use search but without the pagination). Then we might do a premature return from paginate, and then end up returning nil from the `call` method. So it's better to be explicit in the return here I think.

```suggestion
      paginate!
      
      @tasks
```

Reply:
Agreed. I was just copying what other commands did :D